packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -294,6 +303,21 @@ const SceneView = ({
       onHeaderBackButtonClicked={onHeaderBackButtonClicked}
       preventNativeDismiss={isRemovePrevented} // on iOS
       onNativeDismissCancelled={onNativeDismissCancelled}
+      // @ts-expect-error this prop is available since rn-screens 3.26
+      // Unfortunately, because of the bug that exists on Fabric, where native event drivers
+      // for Animated objects are being created after the first notifications about the header height
+      // from the native side, `onHeaderHeightChange` event does not notify
+      // `animatedHeaderHeight` about initial values on appearing screens at the moment.
+      onHeaderHeightChange={Animated.event(
+        [
+          {
+            nativeEvent: {
+              headerHeight: animatedHeaderHeight,
+            },
+          },
+        ],
+        { useNativeDriver: true }
```

Comment:
In cases other than large title, we should add a listener here so we can provide correct value in useHeaderHeight as well. But can be done in a follow up.

Reply:
Won't it break the purpose of the useHeaderHeight? For me `useHeaderHeight` hook is great when you need to have the values that are being updated synchronously and when your app has heavy components that cannot use `useAnimatedHeaderHeight` because of the amount of component updates that uses this hook. In react-native-screens we also have a note about it: `Please beware of using this hook in heavy components, as it may result in performance issues.`.

If user wants to have direct access to the dynamic header's height I would suggest to use `useReanimatedHeaderHeight`, as it uses SharedValue, rather than `useAnimatedHeaderHeight` and going for a listener there. We also have some plans for updating header's height synchronously, but everything will be in its time 😁 

What do you think about it?

Reply:
I'm talking about the header without large title, so there won't be too many updates.

The reason is that the value is hardcoded in JS side for useHeaderHeight and not the actual value. With a listener we should be able to get proper value.

Reply:
Oh, okay I understand 😄 
Then I would suggest to do this as a followup. I'll need to check if `headerHeight` prop in NativeStackView should be changed in any way because of that and furthermore I'll do a refactor or getting status bar height that is being used by `getDefaultHeaderHeight` method 👍

Reply:
Sounds good